### PR TITLE
kubernetes: misc fixes

### PR DESCRIPTION
**Pull request by zowoq**
###### Motivation for this change

- don't install unnecessary binaries

~1100MB -> ~700MB

- ~~conversion-gen~~
- ~~deepcopy-gen~~
- ~~defaulter-gen~~
- e2e.test
- ~~gendocs~~
- ~~genkubedocs~~
- ~~genman~~
- ~~genyaml~~
- ~~go-bindata~~
- ~~go2make~~
- kube-addons
- kube-apiserver
- kube-controller-manager
- kube-proxy
- kube-scheduler
- kubeadm
- kubectl
- kubelet
- mk-docker-opts.sh
- ~~openapi-gen~~
- ~~prerelease-lifecycle-gen~~

Also:

- remove go-bindata, already vendored by upstream
- use installShellFiles, add kubeadm completion
- require kubeadm/kubectl so installPhase can't be broken

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zowoq** [comment]
Semi-related question: Is there a reason we have [`mk-docker-opts.sh`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/cluster/kubernetes/mk-docker-opts.sh) checked in here instead of using the one that is included with [`flannel`](https://github.com/coreos/flannel/blob/master/dist/mk-docker-opts.sh)?

File: pkgs/applications/networking/cluster/kubernetes/default.nix [hunk 1]
```diff
@@ -41,39 +46,46 @@ stdenv.mkDerivation rec {
...
+  ] ++ components);
 
   postBuild = ''
     ./hack/update-generated-docs.sh
     (cd build/pause && cc pause.c -o pause)
   '';
 
   installPhase = ''
-    mkdir -p "$out/bin" "$out/share/bash-completion/completions" "$out/share/zsh/site-functions" "$man/share/man" "$pause/bin"
+    for p in $WHAT; do
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/kubernetes/default.nix:60]: Can we use a different name instead of WHAT?

Comment (zowoq) [reply pkgs/applications/networking/cluster/kubernetes/default.nix:60]: `WHAT` is from the upstream makefile.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105619` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>kubectl</li>
    <li>kubernetes</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105619` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>kubectl</li>
    <li>kubernetes</li>
  </ul>
</details>

**Review by saschagrunert — APPROVED** [review]
LGTM

**Pull request merged**